python_modules/libraries/dagster-deltalake/dagster_deltalake_tests/test_delta_table_resource.py

```diff
@@ -62,3 +63,24 @@ def read_table(delta_table: DeltaTableResource):
             )
         },
     )
+
+
+@pytest.mark.parametrize(
+    "config",
+    [
+        LocalConfig(),
+        AzureConfig(account_name="test", use_azure_cli=True),
+        GcsConfig(bucket="test"),
+        S3Config(bucket="test"),
+        ClientConfig(timeout=1),
+    ],
+)
+def test_resource_storage_options(tmp_path, config):
+    data = pa.table(
+        {
+            "a": pa.array([1, 2, 3], type=pa.int32()),
+            "b": pa.array([5, 6, 7], type=pa.int32()),
+        }
+    )
+
+    write_deltalake(os.path.join(tmp_path, "table"), data, storage_options=config.dict())
```

Comment:
hi, I don't understand how is this testing that the returned value of .dict() matches what you intend? is there an expectation that write_deltalake does that validation properly or throws an exception?

Reply:
Hi @mlarose, thanks for the comment. Indeed the expectation is that by passing the config to `storage_options` in `write_deltalake` it validates that the config is passed in such a way that write_deltalake does not throw an exception.

I will make a small addition to do more explicit validation of the dict. Let me know what you think!